File: squares_of_sorted_array/squares_of_sorted_array.py
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        new_arr = [x for x in range(0, len(A))]
        n = len(A)
        left = 0
        right = len(A) - 1

        for i in range(n - 1, -1, -1):
            if abs(A[left]) > abs(A[right]):
                new_arr[i] = A[left] ** 2
                left += 1
            else:
                new_arr[i] = A[right] ** 2
                right -= 1

        return new_arr

File: squares_of_sorted_array/squares_of_sorted_array.java
class Solution {
    public int[] sortedSquares(int[] A) {
        int n = A.length;
        int[] newArr = new int[n];
        int left = 0;
        int right = A.length - 1;

        for (int i = n - 1; i >= 0; i--) {
            if (Math.abs(A[left]) > Math.abs(A[right])) {
                newArr[i] = A[left] * A[left];
                left++;

            } else {
                newArr[i] = A[right] * A[right];
                right--;
            }
        }
        return newArr;
    }
}